components/script/dom/document.rs

```diff
@@ -4208,6 +4183,25 @@ impl Document {
         self.animations.borrow().cancel_animations_for_node(node);
     }
 
+    pub(crate) fn update_animations_and_send_events(&self, can_gc: CanGc) {
```

Comment:
I think we can add a link to https://drafts.csswg.org/web-animations-1/#update-animations-and-send-events and perhaps document the steps. 

Also step 3 says to perform a microtask checkpoint, which is quite an important TODO I think so we might as well note it here.

Reply:
Thanks for pointing this out. I've gone ahead and added the microtask checkpoint here and updated all of the comments to reflect the specification.